src/govuk/components/pagination/pagination.yaml

```diff
@@ -0,0 +1,169 @@
+params:
+- name: previous
+  type: object
+  required: false
+  description: A link to the previous page, if there is one.
+  params:
+  - name: text
+    type: string
+    required: false
+    description: The previous page link text. Defaults to 'Previous page', where 'page' is visually hidden.
+  - name: label
+    type: string
+    required: false
+    description: Optional label that goes underneath the previous page link providing further context for where it goes.
+  - name: href
+    type: string
+    required: true
+    description: The previous page URL.
+- name: next
+  type: object
+  required: false
+  description: A link to the next page, if there is one.
+  params:
+  - name: text
+    type: string
+    required: false
+    description: The next page link text. Defaults to 'Next page', where 'page' is visually hidden.
+  - name: label
+    type: string
+    required: false
+    description: Optional label that goes underneath the next page link providing further context for where it goes.
+  - name: href
+    type: string
+    required: true
+    description: The next page URL.
+- name: type
+  type: string
+  required: false
+  description: The type of navigation. Goes in the aria label for the encapsulating `nav` element. Defaults to "results".
+- name: items
+  type: array
+  required: false
+  description: Array of link objects.
+  params:
+  - name: number
+    type: number
+    required: false
+    description: The link text - usually a page number.
+  - name: href
+    type: string
+    required: false
+    description: The link URL.
+  - name: current
+    type: boolean
+    required: false
+    description: Set to 'true' to indicate the current page.
+  - name: ellipsis
+    type: boolean
+    required: false
+    description: For specifying if you want an ellipsis at a given point between numbers. If you set this as true, the other attributes aren't required for the same item.
```

Comment:
```suggestion
    description: For specifying if you want an ellipsis at a given point between numbers. If you set this option as `true`, the other attributes are not required for the same item.
```

Reply:
When I say the same item I meant the instance of the pagination item, as part of the `items` array within the pagination macro, that the (presumed) dev will be constructing to call the macro. To illustrate this, here's what a normal pagination `item` might look like:

```
- number: '1'
  href: '/page/1'
```

...whereas here's what an ellipsis `item` looks like:

```
- ellipsis: true
```

What we're trying to convey here is that if the `ellipsis` attribute is set in the macro call of an `item`, then anything that get's passed to that one `item` call will be ignored and so it's a waste of code. For example if we did something like this:

```
- number: '1'
  href: '/page/1'
  ellipsis: true
```

...then the `number` and `href` would be a waste of code as they will be ignored during rendering in favour of just rendering an ellipsis list item. That also hopefully answers your other question (not annoying) that the other attributes are the other `item` params: `number`, `href`, `current` etc.

Does that make sense? I'm wondering if the confusion comes from the fact that I'm using `attributes` to describe thing that go into a macro call and not `params`. Still getting used to our tech docs etymology.

Reply:
That's a great explanation - thanks! Re 'attributes', actually I thinking you're using the right term, that's [what we say throughout the component macro options](https://www.google.com/search?q=%3Csite%3Adesign-system.service.gov.uk%2F+%22attributes%22%3E&rlz=1C5CHFA_enGB970GB970&oq=%3Csite%3Adesign-system.service.gov.uk%2F+%22attributes%22%3E&aqs=chrome..69i57j69i64.4296j0j4&sourceid=chrome&ie=UTF-8).

Reply:
I've applied Ollie's recommendation. I agree that "option" is a suitable word and if we want to create consistency, I'm all for it.